Source/player.cpp

```diff
@@ -2634,12 +2601,15 @@ bool PM_DoAttack(int pnum)
 		PlaySfxLoc(PS_SWING, player.position.tile);
 	}
 
+	bool didhit = false;
+
 	if (player.AnimInfo.CurrentFrame == player._pAFNum) {
 		Point position = player.position.tile + player._pdir;
-		dx = position.x;
-		dy = position.y;
+		int dx = position.x;
+		int dy = position.y;
 
 		if (dMonster[dx][dy] != 0) {
+			int m = -1;

```

Comment:
The code later checks for the value being correctly set by via `m >= 0 && m < MAX`, previously this could fail with the value not being initialized and thus potentially being a random in range value.